base/iterators.jl

```diff
@@ -1595,4 +1595,72 @@ end
 # be the same as the keys, so this is a valid optimization (see #51631)
 pairs(s::AbstractString) = IterableStatePairs(s)
 
+"""
+    nth(itr, n::Integer)
+
+Get the `n`th element of an iterable collection. Return `nothing` if not existing.
```

Comment:
Returning `nothing` makes it impossible to distinguish between "the nth element was `nothing`", and "there was no nth element". Perhaps return `Union{Nothing, Some}`?

Reply:
Fair point.
Should it be `Union{nothing, Some}` even in those cases where we know there can't be a `nothing` value in the iterator? I.e. `Count` Iterator or `Repeated` (with its element different than nothing) or `AbstractRanges`?

Reply:
I think it should, otherwise it would be too confusing.

Reply:
I would just throw an error if there is no `n`th element. There could also be a `default` argument as in `get`, where a user can pass a value that should be returned if no `n`th element exists.

I don't really follow the logic that the spirit of iterators is to return `nothing` in such cases?

Reply:
Agree `nothing` is weird, your iterator can produce that. `Some` seems a bit technical & unfriendly? An error seems fine.

I suppose it can't literally be a method of `get` since it goes by enumeration not keys:
```
julia> first(Dict('a':'z' .=> 'A':'Z'), 3)
3-element Vector{Pair{Char, Char}}:
 'n' => 'N'
 'f' => 'F'
 'w' => 'W'

julia> nth(Dict('a':'z' .=> 'A':'Z'), 3)
'w' => 'W'
```